design/use_cases/indoor_localization_use_case.md

```diff
@@ -0,0 +1,15 @@
+# Indoor Localization
+As a Robot user I want my robot to know its location on a given map of an indoor area so that it can move around the area
+
+## More details
+- Why is this needed?
+   - This is needed for indoor robot navigation in most (all?) cases
+   - Example: a courier robot in a logistics warehouse
+
+- What is the expected user interaction?
+   - The user should be able to specify a map to use and a location on that map for the robot
+   - The robot should be able to deduce it's own position on a map autonomously
+
+- Are there any non-functional requirements? (build system, tools, performance, etc)
+
```

Comment:
Related to my comment above about collision avoidance, what does the user expect the robot to do when it get's lost.

Reply:
Probably alert the Robot User so the user could rectify the situation. Would the robot retrace its steps to a point that it wasn't lost? If so, what would prevent it from following the same path again?